Modify SnapshotPurge logic in longhorn-manager

ttpcodes opened the pull request:
This PR makes changes to the `SnapshotPurge` logic to account for the `SnapshotPurge` operation being moved to the `SyncAgentServer` in `longhorn-engine` as part of longhorn/longhorn-engine#402:
- Various comments and logs have been modified to indicate the `asynchronous` nature of the `SnapshotPurge` operation.
- The `EngineStatus` now stores the `SnapshotPurgeStatus` of the `Volume` during the `refresh` operation.
- The `SnapshotPurge` call during the `RecurringJob` now waits for the `SnapshotPurge` operation to complete before finishing.

This PR implements the `longhorn-manager` portion of longhorn/longhorn#665.

yasker commented on app/volume.go:
```diff
@@ -30,6 +30,8 @@ const (
 	FlagRetain       = "retain"
 	FlagBackupTarget = "backuptarget"
+
+	SnapshotPurgeStatusInterval = 1 * time.Second
```
5 seconds is good enough.

ttpcodes commented:
- Renamed most `fields` to `PurgeStatus` from `SnapshotPurgeStatus`.
- Changed `SnapshotPurgeStatus` polling interval to `5 seconds` in the `Recurring Job`.
- Modified the `SnapshotPurge` call to use the `--skip-if-in-progress` flag.

yasker commented:
LGTM. Wait for the engine part and the integration test part.

ttpcodes commented:
- Exposed `State` field of `PurgeStatus` in `API`.
